### PR TITLE
CXSPA-1322 - added secure portal config initializer

### PR DESCRIPTION
**Pull request by Dalvir-Personal**

### CODE REVIEW

**Comment by giancorderoortiz** [comment]
Please add e2e tests for this new functionality.  @KateChuen fyi.

**Comment by Dalvir-Personal** [comment]
> Please add e2e tests for this new functionality. @KateChuen fyi.

Done

File: projects/core/src/routing/routing.module.ts [hunk 1]
```diff
@@ -54,6 +67,12 @@ export class RoutingModule {
           deps: [ConfigurableRoutesService],
           multi: true,
         },
+        {
+          provide: CONFIG_INITIALIZER,
+          useFactory: initRoutingConfig,
```
Comment (WeizhengSap) [inline projects/core/src/routing/routing.module.ts]: how about rename it to `initSecurePortalConfig`?

File: projects/core/src/routing/configurable-routes/secure-portal-config/secure-portal-config-initializer.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { ConfigInitializer } from '../../../config/config-initializer/config-initializer';
+import { BaseSite } from '../../../model/misc.model';
+import { JavaRegExpConverter } from '../../../util/java-reg-exp-converter/java-reg-exp-converter';
+import { WindowRef } from '../../../window/window-ref';
+import { BaseSiteService } from '../../../site-context/facade/base-site.service';
+import { RoutingConfig } from '../config/routing-config';
+
+@Injectable({ providedIn: 'root' })
+export class SecurePortalConfigInitializer implements ConfigInitializer {
+  readonly scopes = ['context'];
```
Comment (WeizhengSap) [inline projects/core/src/routing/configurable-routes/secure-portal-config/secure-portal-config-initializer.ts]: scopes is `routing`

File: projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/my-company/config/basesites.config.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
+export const baseSitesConfig = {
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/my-company/config/basesites.config.ts]: why we put this config inside "my-company"? I don't think we need it. We can just create a mock baseSiteConfig in b2b-secure-portal.ts, which only contains one site with properties `uid`, `urlPatterns`,`requiresAuthentication: true,`

File: projects/core/src/routing/routing.module.ts (hunk 1 above)
Comment (Dalvir-Personal) [reply projects/core/src/routing/routing.module.ts]: Done

**Review by WeizhengSap — DISMISSED** [review]

**Comment by WeizhengSap** [comment]
I approved the PR, but don't merge until the CI server is updated.

**Review by giancorderoortiz — APPROVED** [review]
Approved based on Wei's previous approval.

**Pull request merged**